tracer/src/Datadog.Trace/AppSec/ControllerContextExtensions.Framework.cs

```diff
@@ -42,12 +41,12 @@ internal static void MonitorBodyAndPathParams(this IControllerContext controller
                 }
 
                 var scope = SharedItems.TryPeekScope(context, peekScopeKey);
-                security.InstrumentationGateway.RaiseBodyAvailable(context, scope.Span, bodyDic);
-                security.InstrumentationGateway.RaisePathParamsAvailable(context, scope.Span, pathParamsDic);
-                security.InstrumentationGateway.RaiseBlockingOpportunity(context, scope, Tracer.Instance.Settings, args =>
+                var securityTransport = new Transports.SecurityTransport(security, context, scope.Span);
```

Comment:
Creating a new `SecurityTransport` object every request seems like it could have a hit on allocations? Have you checked the throughtput tests? Is there any way we can make this a `readonly struct`, or a static method returning a `readonly struct` perhaps? 🤔 I guess it's only hit when appsec is enabled, and we know there's a big memory hit there, so maybe it doesn't matter?

Reply:
Besides the performance implications, I'm not really a fan of we create an instances of this class in a lot different places and call a few different methods in each cases. I think the way the methods have been divided up in SecurityTransport is good, but perhaps we're missing a wrapper class to simplify the interface for client code.

Reply:
Throughput tests seem actually much better 
![image](https://user-images.githubusercontent.com/8353486/203333135-a336083a-85bb-4ed9-9ffd-fe981db089e7.png)

Regarding the allocations on the benchmarks, it's because the refactoring broke the framework benchmarks and the fix is embedding more treatment than before, I'll try to fix this

Reply:
fixed in ef58dfbb0ad8fc4f3297dac3eca7eb91655b72ea as readonly struct and added static helper methods

Reply:
added a set of helpers for the Core part in 5b96f56a0fb40b412b5565a1a6a5d55b2b6d0179, for the framework part, as scenarios are very diverse (combine different addresses dependin on where it is,, cleanup in the end or not) I left it at calling individual methods, as I couldnt see a `CheckPathParamsAndBaseRequestAndCleanup()` , `CheckBodyAndBaseRequest()`, `CheckBodyAndPathParams()`.. 
`SecurityCoordinator` is now a struct and is only created when we check that request is not already blocked